packages/core/lib/middleware/mw_utils.d.ts

```diff
@@ -0,0 +1,56 @@
+import * as http from 'http';
+import Segment = require('../segments/segment');
+import IncomingRequestData = require('./incoming_request_data');
+
+export const defaultName: string | undefined;
+
+export const dynamicNaming: boolean;
+
+export const hostPattern: string | null;
+
+export function enableDynamicNaming(hostPattern?: string): void;
+
+export function processHeaders(req?: Partial<Pick<http.IncomingMessage, 'headers'>>): { [key: string]: string };
+
+export function resolveName(hostHeader?: string): string;
+
+export function resolveSampling(amznTraceHeader: object, segment: Segment, res: http.ServerResponse): void;
```

Comment:
Shouldn't the return type of `resolveSampling()` be a `boolean`?

Reply:
1. That sounds right, I'll update.
2. I don't think `https.IncomingMessage` exists? It's not in `@types/node`, and I don't see it in the source or docs https://nodejs.org/api/https.html
3. The JSDoc does say `@returns {boolean}`, but I think `void` matches the implementation: there is no `return` statement.